File: src/main/java/no/nav/syfo/util/OIDCUtil.java
package no.nav.syfo.util;

import lombok.extern.slf4j.Slf4j;
import no.nav.security.oidc.OIDCConstants;
import no.nav.security.oidc.context.*;

@Slf4j
public class OIDCUtil {

    public static String tokenFraOIDC(OIDCRequestContextHolder contextHolder, String issuer) {
        OIDCValidationContext context = (OIDCValidationContext) contextHolder
                .getRequestAttribute(OIDCConstants.OIDC_VALIDATION_CONTEXT);

        return  context.getToken(issuer).getIdToken();
    }

}

File: src/main/kotlin/no/nav/syfo/api/auth/OIDCUtil.kt
package no.nav.syfo.api.auth

import no.nav.security.token.support.core.context.TokenValidationContextHolder
import no.nav.syfo.api.auth.OIDCClaim.JWT_CLAIM_AZP
import no.nav.syfo.api.auth.OIDCClaim.NAVIDENT

object OIDCUtil {
    @JvmStatic
    fun tokenFraOIDC(contextHolder: TokenValidationContextHolder, issuer: String): String {
        return contextHolder.tokenValidationContext.getJwtToken(issuer).tokenAsString
    }

    fun getNAVIdentFraOIDC(contextHolder: TokenValidationContextHolder): String? {
        val context = contextHolder.tokenValidationContext
        return context.getClaims(OIDCIssuer.VEILEDER_AZURE_V2).getStringClaim(NAVIDENT)
    }

    fun getConsumerClientIdFraOIDC(contextHolder: TokenValidationContextHolder): String? {
        val context = contextHolder.tokenValidationContext
        return context.getClaims(OIDCIssuer.VEILEDER_AZURE_V2).getStringClaim(JWT_CLAIM_AZP)
    }
}
